bittensor/_axon/__init__.py

```diff
@@ -339,101 +339,147 @@ def check_forward_callback( forward_callback:Callable, synapses:list = []):
         forward_callback([sample_input], synapses, hotkey='')
 
 class AuthInterceptor(grpc.ServerInterceptor):
-    """ Creates a new server interceptor that authenticates incoming messages from passed arguments.
-    """
-    def __init__(self, key:str = 'Bittensor',blacklist:List = []):
-        r""" Creates a new server interceptor that authenticates incoming messages from passed arguments.
+    """Creates a new server interceptor that authenticates incoming messages from passed arguments."""
+
+    def __init__(
+        self,
+        wallet: "bittensor.Wallet",
+        blacklist: Callable = None,
+    ):
+        r"""Creates a new server interceptor that authenticates incoming messages from passed arguments.
         Args:
-            key (str, `optional`):
-                 key for authentication header in the metadata (default= Bittensor)
-            black_list (Fucntion, `optional`): 
+            wallet(bittensor.Wallet):
+                the wallet of the receiver which runs the interceptor
+            black_list (Function, `optional`):
                 black list function that prevents certain pubkeys from sending messages
         """
         super().__init__()
-        self._valid_metadata = ('rpc-auth-header', key)
-        self.nounce_dic = {}
-        self.message = 'Invalid key'
+        self.nonces = {}
         self.blacklist = blacklist
-        def deny(_, context):
-            context.abort(grpc.StatusCode.UNAUTHENTICATED, self.message)
-
-        self._deny = grpc.unary_unary_rpc_method_handler(deny)
+        self.wallet = wallet
```

Comment:
I am against putting the whole wallet into server interceptor. The base wallet class contains far too many functionalities and connections for a lightweight interceptor. Instead, this should be an ss58 key.